### PR TITLE
Add Remote Gem repository workflow to Create a Gem workflow

### PR DESCRIPTION
**Pull request by FuzzyCarterAWS**
* Update Create A Gem workflow to support multiple workflows.
* Add workflow for adding a Remote Gem Repository and adding those gems to a project.
* Reformatted **Workflow: Create a Gem from O3DE Project Manager** to utilize a better workflow/requests/things to look for pattern.

### CODE REVIEW

**Comment by LB-KacperKapusta** [comment]
**Workflow 1: Create a Gem from O3DE Project Manager**

**Workflow:** Launch Create a Gem experience
Steps field: 3. Select "Create a New Gem" - actual text in the Project Manager says "Create New Gem"

**Workflow:** Step 1 - Gem Setup Options	
Encountered issues: https://github.com/o3de/o3de/issues/15541

**Workflow:** Step 2 - Gem Details
Expectations field: "All fields can be edited" - could be "All fields can be edited:" for consistency
Encountered issues: https://github.com/o3de/o3de/issues/13736, https://github.com/o3de/o3de/issues/15540

**Workflow:** Created Gem displays in Gem Catalog
Expectations field: The "gem catalog" should be "Gem Catalog" for consistency

The Workflow has been completed with minor issues found.




**Workflow 2: Add a remote gem from O3DE Project Manager**

**Workflow:** Add Remote Gem(s) to Gem Remote Sources Manager.
Encountered issues: https://github.com/o3de/o3de/issues/11159

The Workflow has been completed without major issues.

**Comment by michleza** [comment]
This looks good to me.  Only 2 points things I would suggestion clarifying:

- The remote gems sample referenced at : https://github.com/o3de/sample-code-gems  doesn't have an obvious sign that it's been added to a project after you've built and ran it. I'm wondering if we could put in a request or modify the same gem in some way so that it can be more easily used for testing 
- The stab2305 version of project manager doesn't have a Gem in the ribbon. All I have are Projects or Engine. You can still add gems to the projects via the hamburger button but I'm not sure if I have something incorrectly configured or not

**Comment by FuzzyCarterAWS** [comment]
@michleza 

- [https://github.com/o3de/sample-code-gem](https://github.com/o3de/sample-code-gems) is only a reference example of a repository that has 3 gems (AtomTutorials, ShapeExample, PyShapeExample). Additionally, adding the depot to the gem manager doesn't add the game to a project until after you've actually added the gem to the project (EG: AtomTutorials which will add a bunch of assets and levels to your project, ShapeExample adds a "ShapeExample" editor component entity). If a tester wants to go further, they should make their own repository.
- Diggin in.

**Comment by michleza** [comment]
I think it was a bad build. I uninstalled, downloaded, and reinstalled and the Gems option is there in the ribbon now. LGTM

**Review by michleza — APPROVED** [review]

File: testing-guidance/workflow-tests/o3de-project-manager/create-a-gem/create-a-gem-workflow-tests.md [hunk 1]
```diff
@@ -5,18 +5,102 @@ These workflows center around testing the basic functionality of the Project Man
...
+    ]
+}
+```
+
+**Workflow Requirements**
+* A built project to add a remote gem to (a blank template project is acceptable).
+* Locally installed tools for the remote source being used (git installed with credentials for GitHub, for example)
+* A remote Git repository (GitHub, AWS CodeCommit, BitBucket, self-hosted, ETC) that has a remote project.
+  * Examples: 
+    * sample-code-gems (https://github.com/o3de/sample-code-gems.git)
```
Comment (AMZN-alexpete) [inline testing-guidance/workflow-tests/o3de-project-manager/create-a-gem/create-a-gem-workflow-tests.md:91]: I think we should add the canonical repo too `https://canonical.o3de.org` ([repo.json](https://canonical.o3de.org/repo.json))

**Review by AMZN-alexpete — APPROVED** [review]
Just a minor suggestion to include the canonical o3de repository

File: testing-guidance/workflow-tests/o3de-project-manager/create-a-gem/create-a-gem-workflow-tests.md (hunk 1 above)
Comment (michleza) [reply testing-guidance/workflow-tests/o3de-project-manager/create-a-gem/create-a-gem-workflow-tests.md:91]: I'm not super familiar with remote gems but I like the idea of providing a list of links to some that are commonly used and known to be stable.

**Review by michleza — APPROVED** [review]
LGTM. If you wanted to provide more suggested links to remote gems that would be cool but that's a nit pick.

File: testing-guidance/workflow-tests/o3de-project-manager/create-a-gem/create-a-gem-workflow-tests.md (hunk 1 above)
Comment (FuzzyCarterAWS) [reply testing-guidance/workflow-tests/o3de-project-manager/create-a-gem/create-a-gem-workflow-tests.md:91]: Added

**Pull request merged**